dom-manipulation/main.js

```diff
@@ -0,0 +1,39 @@
+let clickCount = 0;
+
+const $hotButton = document.querySelector(".hot-button")
+const $clickCount = document.querySelector('.click-count');
+
+function incrementClickCount(event) {
+  clickCount += 1;
+}
+
+function updateClickCountDisplay (event) {
+  $clickCount.textContent = 'Clicks: ' + clickCount;
+}
+
+function changeButtonColor(event) {
+  switch (true) {
+    case clickCount < 4:
+      $hotButton.className = 'hot-button cold';
+      break;
+    case clickCount < 7:
+      $hotButton.className = 'hot-button cool';
+      break;
+    case clickCount < 10:
+      $hotButton.className = 'hot-button tepid';
+      break;
+    case clickCount < 13:
+      $hotButton.className = 'hot-button warm';
+      break;
+    case clickCount < 16:
+      $hotButton.className = 'hot-button hot';
+      break;
+    default:
+      $hotButton.className = 'hot-button nuclear';
+      break;
+  }
+}
+
+$hotButton.addEventListener('click', incrementClickCount);
+$hotButton.addEventListener('click', updateClickCountDisplay);
+$hotButton.addEventListener('click', changeButtonColor);
```

Comment:
You don't want to add multiple of the same event listener to an element. It should only have 1 event listener attached to it for each type of event. So all of these function calls should be happening inside of one function that's attached to the event listener